geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/stats/StatsCombiner.scala

```diff
@@ -44,27 +44,40 @@ class StatsCombiner extends Combiner with LazyLogging {
 
   override def reduce(key: Key, iter: java.util.Iterator[Value]): Value = {
     val head = iter.next()
-    if (!iter.hasNext) {
-      head
-    } else {
+    if (!iter.hasNext) { head } else {
       val sftName = try {
         CachedLazyMetadata.decodeRow(key.getRow.getBytes, separator)._1
       } catch {
         // back compatible check
         case NonFatal(e) => SingleRowAccumuloMetadata.getTypeName(key.getRow)
       }
       val serializer = serializers(sftName)
-      val stat = serializer.deserialize(head.get)
+
+      var stat = deserialize(head, serializer)
+
+      while (stat == null) {
+        if (iter.hasNext) {
+          stat = deserialize(iter.next, serializer)
+        } else {
+          return head // couldn't parse anything... return head value and let client deal with it
+        }
+      }
+
       iter.foreach { s =>
-        try {
-          stat += serializer.deserialize(s.get)
-        } catch {
-          case e: Exception => logger.error("Error combining stats:", e)
+        try { stat += serializer.deserialize(s.get) } catch {
```

Comment:
Did you intend to use the new deserialize function here?

Reply:
no, we'd still need a try/catch around the `+=`, and we'd have to check for null. it's a bit redundant, but was trying not to have extra checks